src/history.ts

```diff
@@ -250,12 +252,60 @@ export class HistoryStorage {
     // getting failed history records
     this.failedHistory = await this.db.getAll(TX_FAILED_TABLE);
 
-    console.log(`HistoryStorage: preload ${this.currentHistory.size} history records, ${this.unparsedMemo.size} + ${this.unparsedPendingMemo.size}(pending) unparsed memos(from index ${this.syncIndex + 1})`);
+    // getting failured memo list and include it in unprocessed one
+    // (memos which are placed below syncIndex but wasn't converted to the HistoryRecord for any reason)
+    const failSyncRecords: number[] = await this.db.get(HISTORY_STATE_TABLE, 'fail_indexes');
+    if (failSyncRecords) {
+      await failSyncRecords.forEach(async (idx) => {
```

Comment:
I don't think await works together with forEach
```suggestion
      await Promise.all(failSyncRecords.map(async (idx) => {
```

Reply:
Agree. Fixed in https://github.com/zkBob/zkbob-client-js/pull/97/commits/8c88d1efaadbea8678f406c37da7878d08bd957e